homeassistant/components/rest_command/__init__.py

```diff
@@ -118,13 +124,30 @@ async def async_service_handler(service: ServiceCall) -> None:
                 headers[hdrs.CONTENT_TYPE] = content_type
 
             try:
+                response: ClientResponse
                 async with getattr(websession, method)(
                     request_url,
                     data=payload,
                     auth=auth,
                     headers=headers,
                     timeout=timeout,
                 ) as response:
+                    return_data: ServiceResponse = (
+                        {
+                            "status": response.status,
+                            "headers": [
+                                {"key": item[0], "value": item[1]}
+                                for item in response.headers.items()
+                            ],
+                            "content": (await response.content.read()).decode(
+                                response.get_encoding()
+                            ),
+                            "url": str(response.url),
+                        }
+                        if service.return_response
```

Comment:
Not a fan of inline if-statements, it's hard to read.

Reply:
Fixed.